Fix errors reported by linter.

In #179 I have made it so a linter can be run against the codebase without too many false positives.  We do have some problems that look like they are real:

```
./xdress/autodescribe.py:1961:41: F821 undefined name '_type_error_msg'
./xdress/autodescribe.py:1964:41: F821 undefined name '_type_error_msg'
./xdress/cythongen.py:317:26: F821 undefined name 'name'
./xdress/cythongen.py:1020:84: F821 undefined name 'a'
./xdress/cythongen.py:1023:43: F821 undefined name 'a'
./xdress/typesystem.py:2289:26: F821 undefined name 'base_types'
```

--- a/xdress/autodescribe.py
+++ b/xdress/autodescribe.py
@@ -1958,11 +1958,11 @@ class PycparserVarDescriber(PycparserBaseDescriber):
             for child_name, child in self._root.children():
                 if getattr(child, 'name', None) == self.name:
                     if isinstance(child, pycparser.c_ast.FuncDef):
-                        raise TypeError(_type_error_msg.format(self.name, 'function',
-                                                            'PycparserFuncDescriber'))
+                        raise TypeError(self._type_error_msg.format(
+                            self.name, 'function', 'PycparserFuncDescriber'))
                     if isinstance(child, pycparser.c_ast.Struct):
-                        raise TypeError(_type_error_msg.format(self.name, 'struct',
-                                                            'PycparserClassDescriber'))
+                        raise TypeError(self._type_error_msg.format(
+                            self.name, 'struct', 'PycparserClassDescriber'))
                     self.desc['type'] = self.type(child)
                     break
         else:

--- a/xdress/cythongen.py
+++ b/xdress/cythongen.py
@@ -314,7 +314,6 @@ def funccpppxd(desc, exceptions=True, ts=None):
     cpppxd = _cpppxd_func_template.format(**d)
     extra = desc['extra']
     if 'srcpxd_filename' not in extra:
-        ext = _lang_exts[name.language]
         extra['srcpxd_filename'] = '{0}_{1}.pxd'.format(d['name']['tarbase'])
     return cimport_tups, cpppxd
 
@@ -1017,10 +1016,11 @@ def _gen_argfill(args, defaults):
         elif kind is Arg.VAR:
             afillval = "{0}={1}".format(name, default)
         elif kind is Arg.TYPE:
-            raise ValueError("default argument value cannot be a type: {0}".format(a))
+            raise ValueError("default argument value cannot be a type: "
+                             "{0}".format(name))
         else:
             raise ValueError("default argument value cannot be determined: "
-                             "{0}".format(a))
+                             "{0}".format(name))
         afill.append(afillval)
     return ", ".join(afill), names
 

--- a/xdress/typesystem.py
+++ b/xdress/typesystem.py
@@ -2286,7 +2286,7 @@ class TypeSystem(object):
         """This function will remove a previously registered class from
         the type system.
         """
-        isbase = name in base_types
+        isbase = name in self.base_types
         if not isbase and name not in self.template_types:
             _raise_type_error(name)
         if isbase:

